flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/ResultPartitionTest.java

```diff
@@ -822,6 +822,55 @@ public void testBufferSizeNotChanged() throws IOException {
         assertEquals(bufferSize, subpartition1.pollBuffer().buffer().getSize());
     }
 
+    @Test
+    public void testNumBytesProducedCounterForPipelinedUnicast() throws IOException {
```

Comment:
The change does not differentiate with ResultPartitionType, so I think there is not need to add different tests against different ResultPartitionTypes.

Reply:
I will simplify them.